// hmpps-court-cases-release-dates-design/hmpps/utils/utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatLengths = exports.hmppsFormatDate = exports.personStatus = exports.personDateOfBirth = exports.personProfileName = exports.dayMonthYearForwardSlashSeparator = exports.lastNameCommaFirstName = exports.firstNameSpaceLastName = exports.nameCase = exports.sentenceCase = void 0;
const date_fns_1 = require("date-fns");
const uniformWhitespace = (word) => (word ? word.trim().replace(/\s+/g, ' ') : '');
const sentenceCase = (word) => {
    const uniformWhitespaceWord = uniformWhitespace(word);
    return uniformWhitespaceWord.trim().length >= 1
        ? uniformWhitespaceWord[0].toUpperCase() + uniformWhitespaceWord.toLowerCase().slice(1)
        : '';
};
exports.sentenceCase = sentenceCase;
const nameCase = (name) => {
    const uniformWhitespaceName = uniformWhitespace(name);
    return uniformWhitespaceName
        .split(' ')
        .map(s => (s.includes('-') ? s.split('-').map(exports.sentenceCase).join('-') : (0, exports.sentenceCase)(s)))
        .join(' ');
};
exports.nameCase = nameCase;
const firstNameSpaceLastName = (person) => {
    return `${(0, exports.nameCase)(person.firstName)} ${(0, exports.nameCase)(person.lastName)}`.trim();
};
exports.firstNameSpaceLastName = firstNameSpaceLastName;
const lastNameCommaFirstName = (person) => {
    return `${(0, exports.nameCase)(person.lastName)}, ${(0, exports.nameCase)(person.firstName)}`.replace(/(^, )|(, $)/, '');
};
exports.lastNameCommaFirstName = lastNameCommaFirstName;
const dayMonthYearForwardSlashSeparator = (dateString) => {
    if (!dateString)
        return '';
    const date = (0, date_fns_1.parse)(dateString, 'yyyy-MM-dd', new Date());
    return date && (0, date_fns_1.isValid)(date) ? (0, date_fns_1.format)(date, 'dd/MM/yyyy') : dateString;
};
exports.dayMonthYearForwardSlashSeparator = dayMonthYearForwardSlashSeparator;
const personProfileName = (person) => (0, exports.lastNameCommaFirstName)(person);
exports.personProfileName = personProfileName;
const personDateOfBirth = (dateOfBirth) => (0, exports.dayMonthYearForwardSlashSeparator)(dateOfBirth);
exports.personDateOfBirth = personDateOfBirth;
const personStatus = (status) => (0, exports.sentenceCase)(status);
exports.personStatus = personStatus;
const hmppsFormatDate = (dateString, pattern) => {
    if (!dateString)
        return '';
    const date = (0, date_fns_1.parseISO)(dateString);
    return date && (0, date_fns_1.isValid)(date) ? (0, date_fns_1.format)(date, pattern) : dateString;
};
exports.hmppsFormatDate = hmppsFormatDate;
const formatLengths = (lengths) => {
    if (lengths) {
        const allTimeUnits = ['years', 'months', 'weeks', 'days'];
        const missingTimeUnits = allTimeUnits.filter(timeUnit => !lengths.periodOrder.includes(timeUnit));
        const recordedLength = lengths.periodOrder.reduce((prev, current) => `${prev} ${lengths[current]} ${current}`, '');
        const missingLengths = missingTimeUnits.reduce((prev, current) => `${prev} 0 ${current}`, '');
        return `${recordedLength}${missingLengths ? ` ${missingLengths}` : ''}`;
    }
    return null;
};
exports.formatLengths = formatLengths;
